toc/rfc/rfc-draft-add-support-for-file-based-service-binding.md

```diff
@@ -0,0 +1,112 @@
+- The environment variable length has a hard limit of `131072` bytes per variable which is controlled by the underlying [Linux kernel](https://github.com/torvalds/linux/blob/master/include/uapi/linux/binfmts.h). The environment variable size is defined by `MAX_ARG_STRLEN`, which is a constant with the value `PAGE_SIZE*32` where page size is `4096` bytes. This means that to change the size limit for the CF platform a recompiled kernel with updated value for `MAX_ARG_STRLEN` is required. This limit could be an issue for applications using many services. If the limit is reached by an application, it will fail to stage as discussed in [this issue](https://github.com/cloudfoundry/garden-runc-release/issues/160).
+- Updates of the Service Binding Information require restage. This is not optimal for an eventual support of [Binding rotation](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#binding-rotation) specification from the OSBI API spec.
+
+
+## Proposal
+
+The CFF community should implement an alternative approach for service binding information based on `tmpfs` file(s). Using a file or files to provide service binding information to applications will address the challenges listed in the problem section because:
+- The file size limit can be controlled by the CF platform
+- Already today CF uses `tmpfs` for [Instance Identity Credentials](https://docs.cloudfoundry.org/devguide/deploy-apps/instance-identity.html) which are rotated without restarting the application every `24h` by default
+
+The two approaches should be supported in parallel. Users should be able to select which approach Cloud Controller should use to deliver the binding information.
+There are two alternatives regarding service binding file organization:
+1. The VCAP_SERVICES content is stored in a file which location is specified via the VCAP_SERVICES_FILE_PATH env var in the same format as the VCAP_SERVICES environment variable
+   * Advantages:
+      * Less disruptive for applications consuming the `VCAP_SERVICES` env var
+      * Less implementation effort for the Cloud Controller
+   * Disadvantages:
+      * Can’t make use of tools and libraries from the Cloud Native community because K8s specifies a different file structure and format for the service binding information
+2. Implement the K8s service binding specification. The environment variable `SERVICE_BINDING_ROOT` defines the location for the service bindings. There is a file per service binding. The name of the file and the format follow the [K8s specification](https://servicebinding.io/):
+   * Advantages:
+      * CF community could re-use service binding libraries from the Cloud Native community
+      * Moving application between CF or K8s deployments will be easier
+   * Disadvantages
+      * Higher implementation efforts for the Cloud Controller
+      * Applications with binding information >`130KB` have to go with the file option and adopt it
```

Comment:
Isn't that a disadvantage of the whole proposal? I don't think it only applies to alternative 2)

Reply:
Thanks! You are right and I tried to address this with https://github.com/cloudfoundry/community/commit/40681233f3aaad7ff80519e114ff60372ce742b4